Straight-forward changes made by 2to3.py means now 2/3 compatible

RichardWaiteSTFC opened the pull request:
Subset of files in repository for which the changes by 2to3.py script are judged to be minor and low-risk (mostly print functions, range -> list(range) and similar for dict.keys() etc. and some maps have been replaced with list comprehension). In addition I have added a line to import print_function (removed by 2to3.py) to maintain python 2 compatibility. I did this with a script which adds the line to the top of the file (after the first block of comments if present). Note in some cases there are blank lines between comments at the top of the file and so it might not be added where the other imports are.

A code-check has been performed to check the files will run in python 2 and 3 - but this has not been tested.

To test:
(1) Review the code changes
(2)  Test a subset of the scripts to confirm they run in python 2 and 3.
 
In some cases the script may depend on another script for which I haven't committed the changes from python 2 to 3 as they are less trivial/I am not sure they will work. A spreadsheet with a list of files not committed and the reason are attached.

[Python3_mantidScriptRepo.xlsx](https://github.com/mantidproject/scriptrepository/files/3851663/Python3_mantidScriptRepo.xlsx)

RichardWaiteSTFC committed:
Fixed line endings for git diff and slecetively import print-function

Line endings are now the same as in the origional file and import print_function only added to file is print command used

RichardWaiteSTFC committed:
Added a file after check that date function worked post 2to3.py

2to3.py adds "o" to date commands e.g. "0o3" for 03 which works in pyhton 2 and 3

NickDraper commented on BilbyCustomFunctions_Reduction.py:
```diff
@@ -181,7 +182,7 @@ def wavelengh_slices(wavelength_intervals, binning_wavelength_ini, wav_delta):
             n = math.floor((wav2 - wav1)/wav_delta)
             max_wave_length = wav1 + n*wav_delta           
-            print ('\n WARNING: because of your set-up, maximum wavelength to consider for partial reduction is only %4.2f \n' %max_wave_length)
+            print(('\n WARNING: because of your set-up, maximum wavelength to consider for partial reduction is only %4.2f \n' %max_wave_length))
```
This second bracket is probably not needed.

NickDraper commented on direct_inelastic/LET/PolCorr.py:
```diff
@@ -114,8 +115,8 @@ def FAP_gamma(gamma, PF, PHe):
             print("****************************************") 
-            print("PF   = {0:1.3f} +/- {1:1.3f}".format(PF, PFe))
-            print("P_He = {0:1.3f} +/- {1:1.3f}".format(PHe, PHee))
+            print(("PF   = {0:1.3f} +/- {1:1.3f}".format(PF, PFe)))
```
too many brackets?

NickDraper commented on direct_inelastic/LET/PolCorr.py:
```diff
@@ -114,8 +115,8 @@ def FAP_gamma(gamma, PF, PHe):
-            print("PF   = {0:1.3f} +/- {1:1.3f}".format(PF, PFe))
-            print("P_He = {0:1.3f} +/- {1:1.3f}".format(PHe, PHee))
+            print(("PF   = {0:1.3f} +/- {1:1.3f}".format(PF, PFe)))
+            print(("P_He = {0:1.3f} +/- {1:1.3f}".format(PHe, PHee)))
```
too many brackets?

NickDraper commented on direct_inelastic/LET/PolCorr.py:
```diff
@@ -140,7 +141,7 @@ def get_helium_from_ROI(self, min_t, max_t, roi="MaskWorkspace"):
         
         print("\nStarting get_helium_from_ROI...")
-        print("Using {0} meV rep with PF={1}".format(ei,PF))
+        print(("Using {0} meV rep with PF={1}".format(ei,PF)))
```
```suggestion
        print("Using {0} meV rep with PF={1}".format(ei,PF))
```

NickDraper commented on direct_inelastic/LET/PolCorr.py:
```diff
@@ -150,7 +151,7 @@ def get_helium_from_ROI(self, min_t, max_t, roi="MaskWorkspace"):
                 SF_run = run
 
-            print("Calculating P_He from ROI for runs NSF:{0} and SF:{1} at {2:<3.2f}meV ".format(NSF_run,SF_run,ei))
+            print(("Calculating P_He from ROI for runs NSF:{0} and SF:{1} at {2:<3.2f}meV ".format(NSF_run,SF_run,ei)))
```
```suggestion
            print("Calculating P_He from ROI for runs NSF:{0} and SF:{1} at {2:<3.2f}meV ".format(NSF_run,SF_run,ei))
```

NickDraper commented on direct_inelastic/LET/PolCorr.py:
```diff
@@ -208,7 +209,7 @@ def exp_decay(t,T1,P0):
         he_fit_ws   = CreateWorkspace(np.array(times),np.array(he_fit))
 
-        print("\nInitial Cell polarization P0={0:1.3f}+/-{1:1.3f} with lifetime T1={2:3.2f}+/-{3:1.2f}".format(self.PHe0,self.PHe0e,-self.T1,self.T1e))
+        print(("\nInitial Cell polarization P0={0:1.3f}+/-{1:1.3f} with lifetime T1={2:3.2f}+/-{3:1.2f}".format(self.PHe0,self.PHe0e,-self.T1,self.T1e)))
```
```suggestion
        print("\nInitial Cell polarization P0={0:1.3f}+/-{1:1.3f} with lifetime T1={2:3.2f}+/-{3:1.2f}".format(self.PHe0,self.PHe0e,-self.T1,self.T1e))
```

NickDraper commented on direct_inelastic/LET/PolCorr.py:
```diff
@@ -231,7 +232,7 @@ def get_helium_parameters(self):
 
         print("\nStarting get_helium_parameters...")
-        print("Using {0} meV rep with PF={1}".format(ei,PF))
+        print(("Using {0} meV rep with PF={1}".format(ei,PF)))
```
```suggestion
        print("Using {0} meV rep with PF={1}".format(ei,PF))
```

NickDraper commented on direct_inelastic/MARI/MARIReduction_SampleAbs.py:
```diff
@@ -350,7 +351,7 @@ def new_get_workspace(self):
                             for id, ws_out in enumerate(red_ws):
                                 print('--------------------')
-                                print(ws_out.name())
+                                print((ws_out.name()))
```
Yes just more of the print (( ... )), lots of occurances of this that we should clean up.

NickDraper commented on direct_inelastic/MERLIN/example_pychop_script_merlin.py:
```diff
@@ -1,4 +1,5 @@
 #Example script for using PyChop
+from __future__ import print_function
```
No print function calls in this script so we don't need this

RichardWaiteSTFC replied:
There are print statements in this file they just already had brackets but there was no import print_function

NickDraper commented on muon/muesr/examples/Fe_bcc/run_example.py:
```diff
@@ -1,4 +1,5 @@
 #   This is an example to calculate the muon local field at the tetrahedral site(s) in bcc-Fe.
+from __future__ import print_function
```
We should not edit files if they are already 2/3 compatible

NickDraper commented on muon/muesr/examples/LaFeAsO/run_example.py:
```diff
@@ -1,3 +1,4 @@
+from __future__ import print_function
```
We should not edit files if they are already 2/3 compatible

NickDraper commented on muon/muesr/examples/LiFePO4/run_example.py:
```diff
@@ -4,6 +4,7 @@
 will be done first using the helper function which facilitate the user input
 and then programmatically.
 """
+from __future__ import print_function
```
We should not edit files if they are already 2/3 compatible

NickDraper commented on muon/muesr/examples/MnSi/run_example.py:
```diff
@@ -1,3 +1,4 @@
+from __future__ import print_function
```
We should not edit files if they are already 2/3 compatible

NickDraper commented on muon/muesr/examples/test/example.py:
```diff
@@ -1,3 +1,4 @@
+from __future__ import print_function
```
We should not edit files if they are already 2/3 compatible and many other files like this

RichardWaiteSTFC commented on direct_inelastic/LET/PolCorr.py:
```diff
@@ -140,7 +141,7 @@ def get_helium_from_ROI(self, min_t, max_t, roi="MaskWorkspace"):
         
         print("\nStarting get_helium_from_ROI...")
-        print("Using {0} meV rep with PF={1}".format(ei,PF))
+        print(Using {0} meV rep with PF={1}".format(ei,PF))
```
Missing an "

RichardWaiteSTFC commented on direct_inelastic/LET/PolCorr.py:
```diff
@@ -150,7 +151,7 @@ def get_helium_from_ROI(self, min_t, max_t, roi="MaskWorkspace"):
                 SF_run = run
 
-            print("Calculating P_He from ROI for runs NSF:{0} and SF:{1} at {2:<3.2f}meV ".format(NSF_run,SF_run,ei))
+            print(Calculating P_He from ROI for runs NSF:{0} and SF:{1} at {2:<3.2f}meV ".format(NSF_run,SF_run,ei))
```
Also missing an "

RichardWaiteSTFC commented on direct_inelastic/LET/PolCorr.py:
```diff
@@ -208,7 +209,7 @@ def exp_decay(t,T1,P0):
         he_fit_ws   = CreateWorkspace(np.array(times),np.array(he_fit))
 
-        print("\nInitial Cell polarization P0={0:1.3f}+/-{1:1.3f} with lifetime T1={2:3.2f}+/-{3:1.2f}".format(self.PHe0,self.PHe0e,-self.T1,self.T1e))
+        print(\nInitial Cell polarization P0={0:1.3f}+/-{1:1.3f} with lifetime T1={2:3.2f}+/-{3:1.2f}".format(self.PHe0,self.PHe0e,-self.T1,self.T1e))
```
missing an "

RichardWaiteSTFC commented on direct_inelastic/LET/PolCorr.py:
```diff
@@ -231,7 +232,7 @@ def get_helium_parameters(self):
 
         print("\nStarting get_helium_parameters...")
-        print("Using {0} meV rep with PF={1}".format(ei,PF))
+        print(Using {0} meV rep with PF={1}".format(ei,PF))
```
missing an "

RichardWaiteSTFC committed:
Removed double import print_function and double brakets after prints

Only print statements that were the only thing on that line were ammended (done by automated script)

RichardWaiteSTFC committed:
Manual correction of print brackets for unusual cases.

RichardWaiteSTFC committed:
Added line breaks between import and print to some test scripts

NickDraper commented on direct_inelastic/MERLIN/calibration/create_MERInst_Files.py:
```diff
@@ -209,7 +209,7 @@ def check_calibration_options(cycle_end_list):
     else:
         commit_changes = False
-        print(('*** --- Can not find default repository path at {0} so no comitting to public repository will be attempted'.format(repository_path)))                
+        print('*** --- Can not find default repository path at {0} so no comitting to public repository will be attempted'.format(repository_path)))               
```
mismatched brackets here and in a few other places too

NickDraper commented on direct_inelastic/MERLIN/calibration/create_MERInst_Files.py:
```diff
@@ -25,7 +25,7 @@ def create_MERInst_Files(run,commit_changes,repository_path,cycle_index,one2one_
     #
     print ('*************************************************************')
-    print(('*** Creating MERLIN instrument files within working directory: {0}'.format(working_dir)))    
+    print('*** Creating MERLIN instrument files within working directory: {0}'.format(working_dir)))   
```
mismatched brackets

NickDraper commented on direct_inelastic/MERLIN/calibration/create_MERInst_Files.py:
```diff
@@ -227,9 +227,9 @@ def check_calibration_options(cycle_end_list):
 
     print('*** for cycle: {0}'.format(cycle_index))  
-    print(('*** Instrument files will be build in the folder {0}'.format(config['defaultsave.directory'])))                    
+    print('*** Instrument files will be build in the folder {0}'.format(config['defaultsave.directory'])))                   
```
and here

RichardWaiteSTFC committed:
Corrected more mis-matched brackets on print functions in one file

NickDraper commented on muon/CollateBeamCameraSteering.py:
```diff
@@ -252,7 +253,7 @@
 			gradY=fr[3].cell(1,1)
 			DeleteWorkspace("tmpgrad")
-			print (mag,magref,X0,Y0,gradX,gradY)
+			print(mag,magref,X0,Y0,gradX,gradY))
```
mismatched bracket

NickDraper commented on muon/SampleLogCountRateAnalyser4Emu.py:
```diff
@@ -179,15 +180,15 @@ def getproperty(self):
 		det_table.addRow(rnum_detcounts)
 
-		print("Run ",n," processed")
+		print("Run ",n," processed"))
```
mismatched bracket

NickDraper commented on muon/SampleLogCountRateAnalyser4Emu.py:
```diff
@@ -179,15 +180,15 @@ def getproperty(self):
-		print 'args:' + str(e.args)
-		print 'message:' + e.message
-		print 'e itself:' + str(e)
+		print("Run ",n," was not processed"))
```
mismatched braket

RichardWaiteSTFC committed:
Even more mis-matched print brackets corrected

NickDraper reviewed (approved):
Code review is as good as you can reasonably achieve on 144 files
